File: app/src/main/java/org/amoustakos/boilerplate/injection/component/DialogComponent.java
package org.amoustakos.boilerplate.injection.component;


import android.app.Dialog;

import org.amoustakos.boilerplate.injection.PerActivity;
import org.amoustakos.boilerplate.injection.module.DialogModule;

import dagger.Subcomponent;

/**
 * This component injects dependencies to all Dialogs across the application
 */
@PerActivity
@Subcomponent(modules = DialogModule.class)
public interface DialogComponent {

    // =========================================================================================
    // Injections
    // =========================================================================================

    void inject(Dialog dlg);

}

File: app/src/main/java/org/amoustakos/boilerplate/injection/component/DialogComponent.kt
package org.amoustakos.boilerplate.injection.component


import android.app.Dialog
import dagger.Subcomponent
import org.amoustakos.boilerplate.injection.annotations.scopes.PerActivity
import org.amoustakos.boilerplate.injection.module.injectors.DialogModule

/**
 * This component injects dependencies to all Dialogs across the application
 */
@PerActivity
@Subcomponent(modules = [
	DialogModule::class
])
interface DialogComponent {

	fun inject(dlg: Dialog)

}
